### PR TITLE
feat($lineWithFocusPlusSecondary): Fix bugs + add interactive layer

### PR DESCRIPTION
**Pull request by jeznag**
Handle legend issue.
Convert back to ES5.

ELNK-720

### CODE REVIEW

**Comment by jufemaiz** [comment]
Localised viewing via git checkout branch and opening examples/lineWithFocusPlusSecondaryChart.html

1. Onload, secondary chart shows full space, while the focus is highlighting a zoomed area – thus primary + secondary x-axis are not in alignment with the expected focus;
2. Zoom out to entire space (click once in the focus panel outside of the selected area) and both the primary and secondary areas are not showing properly;
3. Deselecting a secondary line via the legend removes it from the legend;
4. Hover doesn't provide a guideline along with values on both axis.

I'd like to see another example with:

* Primary & secondary axis values with different but sometimes coincident datetimestamps
* Point highlighting with values

**Comment by jeznag** [comment]
@jufemaiz can you check again? I think I've fixed those issues.

**Comment by jufemaiz** [comment]
When zooming, the graph "pane" is not filled.
screenshot 2018-02-27 10 21 36

Toggling a line redraws the lines with animations when zoomed (this is different to current experience on the lineWithFocus) – i.e. it goes "white" before rendering all lines again.

That's the last two things I can see as issues.

File: src/models/lineChart.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ nv.models.lineChart = function() {
         , showXAxis = true
         , showYAxis = true
         , rightAlignYAxis = false
-        , useInteractiveGuideline = true
+        , useInteractiveGuideline = false
```
Comment (jufemaiz) [inline src/models/lineChart.js]: Is this changing the defaults used by nvd3 itself?

Comment (jeznag) [reply src/models/lineChart.js]: Nah it's fixing a bug I introduced in our fork (and have fixed in the energylink version of nvd3). nvd3 master has false: https://github.com/novus/nvd3/blob/master/src/models/lineChart.js#L27

File: examples/lineWithFocusPlusSecondaryChart.html [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.2/d3.min.js" charset="utf-8"></script>
     <script src="../build/nv.d3.js"></script>
     <script src="lib/stream_layers.js"></script>
+  <script src="https://code.jquery.com/jquery-3.3.1.js" integrity="sha256-2Kok7MbOyxpgUVvAk/HJ2jigOSYS2auK4Pfzbm7uH60=" crossorigin="anonymous"></script>
```
Comment (jufemaiz) [inline examples/lineWithFocusPlusSecondaryChart.html]: Why jQuery?

**Comment by jufemaiz** [comment]
@jeznag the onload pre-focused is still glitchy. Otherwise it seems the issues are resolved.

**Comment by jeznag** [comment]
@jufemaiz can you check again?

**Comment by jufemaiz** [comment]
Issues related to the focus on load resolved.

Still having issues with examples/lineWithFocusPlusSecondaryChart2.html

screenshot 2018-03-21 08 56 50

**Comment by jeznag** [comment]
Derp - fixed

**Pull request merged**